Profile page (step 1 - 6)

rosafish opened the pull request:
Created profile page for a user that has a "about me" section and a "past messages" section. "about me" can be edited by the user and updates accordingly. "past messages" are displayed by time order of conversation and follows time order within each conversation.

hyang344 committed:
Created user.jsp

Create user.jsp for empty webpage.

hyang344 committed:
Handles user servlets

(Currently) Get the name after the /user/ URI and check whether is valid, then call user.jsp. Does not do post request yet.

hyang344 committed:
Handles user servlet

Accepts all url pattern with /user/*

hyang344 committed:
Include navbar in empty page

As title

hyang344 committed:
Added about me section

(Currently) About me section is empty with white box (just kidding its a pretty good image)

hyang344 committed:
"Edit button" for about me

Only shows the edit button if the viewer is at the same time the page's owner. Now the jsp file knows who is the user & viewer.

hyang344 committed:
Add aboutMe fields

Add getters and setters so the user will be able to access aboutMe and change it

hyang344 committed:
AboutMe portion

Now the aboutMe shows up in the user page. Can be editted if he/she is the user and the submit/cancel button is built. Does not take in user data yet.

rosafish committed:
Update aboutMe after submission in editAboutMe

Takes in user data in editAboutMe, updates and displays the new aboutMe accordingly

rosafish committed:
Past conversation box

created a box to display past conversation of a user on the user's profile page

rosafish committed:
load and display past messages

hard-coded to load and display all past messages of this user in this session.

rosafish committed:
Fixed bug so past messages can show up even at a new session

changed == to .equals when comparing two UUID objects

rosafish committed:
store updated aboutme

store updated aboutme into datastore, but when a new session starts, the aboutme is still not updated

rosafish committed:
display updated aboutme

fixed the issue by creating a new constructor for user that takes in the property aboutme stored in datastore to display aboutme

tsurantino commented on src/main/java/codeu/controller/UserServlet.java:
```diff
@@ -0,0 +1,90 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet class responsible for the conversations page. */
```
This should be updated to refer to the "User" or "Profile" page rather than Conversations page.

tsurantino commented on src/main/webapp/WEB-INF/view/user.jsp:
```diff
@@ -22,10 +22,30 @@ User user = (User) request.getAttribute("user");
+      
+  <div id="container">
+    <div
+      style="width:75%; margin-left:auto; margin-right:auto; margin-top: 50px;">
```
We shouldn't be putting these styling rules in the HTML markup. Can you please move these to a separate CSS file that's referenced on this page?

tsurantino commented on src/main/webapp/WEB-INF/view/user.jsp:
```diff
@@ -22,10 +22,30 @@ User user = (User) request.getAttribute("user");
 <head>
   <title><%= user.getName() %></title>
   <link rel="stylesheet" href="/css/main.css" type="text/css">
+  <style>
```
You should move this to a separate file, ie. main.css

tsurantino commented on src/main/java/codeu/controller/UserServlet.java:
```diff
@@ -0,0 +1,180 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet class responsible for the conversations page. */
```
Nit: but this should say User/Profile, not conversations.

hyang344 replied:
Just changed :)

tsurantino commented on src/main/java/codeu/controller/UserServlet.java:
```diff
@@ -0,0 +1,180 @@
+  /**
+   * This function fires when a user submits a change in the about me section. It gets the
+   * logged-in username from the session and the new conversation title from the submitted form
+   * data. It uses this to create a new Conversation object that it adds to the model.
```
These comments should be updated to reflect what is actually happening.

hyang344 replied:
Just changed :)

tsurantino commented on src/main/java/codeu/controller/UserServlet.java:
```diff
@@ -0,0 +1,180 @@
+		System.out.println("Cancel!");
+	} else if (buttonVal.equals("submit")){
+		editAboutMe = false; // finish editing
+        user.setAboutMe(enteredAboutMe);
```
Indentation looks weird here, could you please update?

hyang344 replied:
Just changed :)

tsurantino commented on src/main/java/codeu/model/store/basic/UserStore.java:
```diff
@@ -104,6 +104,11 @@ public void addUser(User user) {
     persistentStorageAgent.writeThrough(user);
   }
 
+  /** Add a new user to the current set of users known to the application.*/
```
This comment doesn't look like it's describing the function.

tsurantino commented on src/main/java/codeu/model/store/persistence/PersistentDataStore.java:
```diff
@@ -153,8 +154,24 @@ public void writeThrough(User user) {
+    for (Entity entity : results.asIterable()) {
+        UUID uuid = UUID.fromString((String) entity.getProperty("uuid"));
+        if (uuid.equals(user.getId())) {
+          entity.removeProperty("aboutme");
```
Do we necessarily need to remove the property? Doesn't setProperty just override this field?

tsurantino reviewed (changes requested):
Just a few minor nits about comment and naming, otherwise it looks good to go.

hyang344 committed:
Change comments & Indentation

Changes the file into indentation of 4's & make sure the comments make sense.

rosafish committed:
changed comments and persistentDataStore

handled null aboutme, also got rid of removing property aboutme, because setting property updates it
